### PR TITLE
[gRPC] Migrate gcs data structures to protobuf

### PR DESCRIPTION
**Pull request by raulchen**
## What do these changes do?

- Migrate flatbuffers-defined GCS data strucutures to protobuf. Some exceptions that will be handled in the next PR:
  - Task table, which is being depended by a lot of code.
  - Profile table, which is being used by raylet client and is temporarily duplicated in both `gcs.proto` and `gcs.fbs`.
- Remove some deprecated tables.

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Comment by raulchen** [comment]
@zhijunfu @jiangzihao2009

**Comment by raulchen** [comment]
Jenkins, retest this

File: src/ray/raylet/reconstruction_policy.h [hunk 1]
```diff
@@ -17,6 +17,8 @@ namespace ray {
 
 namespace raylet {
 
+using rpc::TaskReconstructionData;
```
Comment (zhijunfu) [inline src/ray/raylet/reconstruction_policy.h]: better do this in .cc file to avoid potential conflicts?

File: src/ray/raylet/raylet.h [hunk 2]
```diff
@@ -16,6 +16,8 @@ namespace ray {
 
 namespace raylet {
 
+using rpc::ClientTableData;
```
Comment (zhijunfu) [inline src/ray/raylet/raylet.h]: can this be done in .cc file instead o avoid potential conflicts?

File: src/ray/raylet/node_manager.h [hunk 3]
```diff
@@ -26,6 +25,13 @@ namespace ray {
 
 namespace raylet {
 
+using rpc::ActorTableData;
+using rpc::ClientTableData;
+using rpc::DriverTableData;
+using rpc::ErrorType;
+using rpc::HeartbeatBatchTableData;
+using rpc::HeartbeatTableData;
```
Comment (zhijunfu) [inline src/ray/raylet/node_manager.h]: is it possible to move these using inside the NodeManager class, or move to .cc file?

File: src/ray/raylet/monitor.h [hunk 4]
```diff
@@ -11,6 +11,10 @@ namespace ray {
 
 namespace raylet {
 
+using rpc::ClientTableData;
+using rpc::HeartbeatBatchTableData;
+using rpc::HeartbeatTableData;
```
Comment (zhijunfu) [inline src/ray/raylet/monitor.h]: to .cc?

**Review by zhijunfu — COMMENTED** [review]
Thanks, looks good to me. Left a few comments.

File: src/ray/raylet/reconstruction_policy.h (hunk 1 above)
Comment (raulchen) [reply src/ray/raylet/reconstruction_policy.h]: Previously, those data structures in `gcs.fbs` were defined in global namspace. 
If I tried putting this in the `.cc` files, but ended up with adding `::rpc` everywhere in all the header files. So I reverted it back. 
It's not ideal, but not worse than before..

File: src/ray/raylet/node_manager.h (hunk 3 above)
Comment (raulchen) [reply src/ray/raylet/node_manager.h]: C++ doesn't seem to allow putting `using` in class definition.

File: python/ray/gcs_utils.py [hunk 5]
```diff
@@ -70,22 +74,9 @@ def construct_error_message(driver_id, error_type, message, timestamp):
...
-    ray.core.generated.ErrorTableData.ErrorTableDataAddErrorMessage(
-        builder, message_offset)
-    ray.core.generated.ErrorTableData.ErrorTableDataAddTimestamp(
-        builder, timestamp)
-    error_data_offset = ray.core.generated.ErrorTableData.ErrorTableDataEnd(
-        builder)
-    builder.Finish(error_data_offset)
-
-    return bytes(builder.Output())
+    data = ErrorTableData()
```
Comment (pcmoritz) [inline python/ray/gcs_utils.py]: oh yes, this is so much better!

**Review by pcmoritz — APPROVED** [review]
Ok, I had a close look through it, it looks good to me. I'll merge it now since otherwise it will generate lots of conflicts soon.

**Pull request merged**